src/lib/auth/safari-authenticator.js

```diff
@@ -0,0 +1,160 @@
+import Authenticator from '@/lib/auth/authenticator.js'
+
+/**
+ * This is a base class for all other authenticator classes.
+ * It defines an API that is common across all variations of authenticator objects.
+ * An authenticator object serves as a proxy between a component's Auth data module
+ * and platform-specific lower level authentication functionality.
+ */
+export default class SafariAuthenticator extends Authenticator {
+  /**
+   * @constructor
+   */
+  constructor () {
+    super()
+
+    // User data props
+    this.isAuthenticated = false
+    this.hasUserData = false
+    // A full name of the user
+    this.userName = null
+    // A user's nickname
+    this.userNickname = null
+    // A user ID
+    this.userId = null
+
+    this.accessToken = null
+
+    // TODO: Hardcoded for now. Can we do it any better than that?
+    this.endpoints = {
+      wordlist: 'https://userapis.alpheios.net/v1/words',
+      settings: 'https://settings.alpheios.net/v1/settings'
+    }
+  }
+
+  /**
+   * A link for login external to the Alpheios components.
+   * @returns null for client side login
+   */
+  loginUrl () {
+    return null
+  }
+
+  /**
+   * A link for logout external to the Alpheios components.
+   * @returns null for client side login
+   */
+  logoutUrl () {
+    return null
+  }
+
+  getEndPoints () {
+    return new Promise((resolve, reject) => {
+      if (this.hasUserData) {
+        // User data has been retrieved
+        resolve(this.endpoints)
+      } else {
+        reject('No user data has been retrieved yet')
+      }
+    })
+  }
+
+  /**
+   * Check access token expiration.
+   * @returns {Promise}
+   */
+  session () {
+    return new Promise((resolve, reject) => {
+      reject('Not implemented')
+    })
+  }
+
+  /**
+   * @typedef SafariAuthData
+   * @property {string} userId - A user ID (`sub` in Auth0).
+   * @property {string} userName - A full name of the user.
+   * @property {string} userNickname - A user's nickname.
+   * @property {string} accessToken - An access token of the user..
+   */
+  /**
+   * Authenticates user with an Auth0.
+   * @param {SafariAuthData} authData - A user auth data.
+   * @returns {Promise}
+   */
+  authenticate (authData) {
+    return new Promise((resolve, reject) => {
+      this.userId = authData.userId
```

Comment:
should we maybe add some checks here, to be sure the authData object is complete and reject the promise if it isn't?

Reply:
I think this is a great idea. Will update